src/IOHandler.java

```diff
@@ -0,0 +1,107 @@
+import java.io.*;
+import java.util.Scanner;
+
+public class IOHandler {
+    private static File file;
+    private static Scanner fileInput;
+
+    // Open a file and keep it open for reading until next close / next change
+    public void OpenFile(String filename) {
+        try {
+            file = new File(filename);
+            fileInput = new Scanner(file);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found: " + filename);
+        }
+    }
+
+    // Code from MadLibs project we did in class
+    public boolean fileHasNextLine() {
+        if (fileInput == null) {
+            return false;
+        } else {
+            return fileInput.hasNextLine();
+        }
+    }
+
+    public String getNextLine() {
+        if (fileHasNextLine()) {
+            return fileInput.nextLine();
+        } else {
+            return "";
+        }
+    }
+
+    public void writeFileLine(String filePath, String line) {
+        // https://www.digitalocean.com/community/tutorials/java-filewriter-example
+        // Most ideas from this article
+        try {
+            FileWriter fileWriter = new FileWriter(filePath, true);
+            PrintWriter printWriter = new PrintWriter(fileWriter);
+            printWriter.println(line);
+            printWriter.close();
+        // Handle IOExceptions
+        // https://docs.oracle.com/javase/8/docs/api/java/io/IOException.html
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Became important for editing seed / flashcards studied
+    // in the deck file
+    public void editFileLine(String filePath, int lineNumber, String newLine) {
+        try {
+            File inputFile = new File(filePath);
+
+            // Creates a temp file to write the new content to
```

Comment:
this feels messy. What if you just read the entire file in as an array of strings, edited the one line, the wrote the file back out? There is a `readAllLines()` method in one one of the java file classes that will read the lines as a list for you